baby-gru/src/components/validation-tools/MoorhenIrisValidation.tsx

```diff
@@ -28,31 +29,71 @@ export const MoorhenIrisValidation = (props: {
     const width = useSelector((state: moorhen.State) => state.sceneSettings.width)
     const height = useSelector((state: moorhen.State) => state.sceneSettings.height)
     const molecules = useSelector((state: moorhen.State) => state.molecules)
+    const maps = useSelector((state: moorhen.State) => state.maps)
+
+    const [result, setResult] = useState();
 
     const random_data = generate_random_data(5) // get 5 metric rings
+    const [selectedModel, setSelectedModel] = useState<null | number>(null)
+    const [selectedMap, setSelectedMap] = useState<null | number>(null)
+    const mapSelectRef = useRef<undefined | HTMLSelectElement>();
+    const moleculeSelectRef = useRef<undefined | HTMLSelectElement>();
+
+    const handleModelChange = (evt: React.ChangeEvent<HTMLSelectElement>) => {
+        setSelectedModel(parseInt(evt.target.value))
+    }
+
+    const handleMapChange = (evt: React.ChangeEvent<HTMLSelectElement>) => {
+        setSelectedMap(parseInt(evt.target.value))
+    }
 
     const aes: IrisAesthetics = {
         dimensions: [plotDimensions, plotDimensions],
-        radius_change: 50, 
+        radius_change: 50,
         header: 40,
-        text_size: 50
+        text_size: 100
     }
 
     const results: IrisData = {
-        data: random_data,
-        chain_list: ["A", "B", "C"],
-        file_list: ["input1"], 
-    } 
-    
-    const iris_props: IrisProps = { 
+        data: result,
+        chain_list: null,
+        file_list: [`${molecules[0].name}.pdb`],
+    }
+
+    const iris_props: IrisProps = {
         results: results,
-        from_wasm: false,
-        aesthetics: aes, 
-        callback: (residue) => { 
-            console.log("RESIDUE CLICKED", residue)
+        from_wasm: true,
+        aesthetics: aes,
+        click_callback: (residue) => {
+            const split_res = residue.split("/")
+            const res = `${split_res[0]}/${split_res[2]}(${split_res[1]})`
+            molecules[0].centreOn(res)
+        },
+        hover_callback: (residue) => {
+            // const split_res = residue.split("/")
+            // const res = `${split_res[0]}/${split_res[2]}(${split_res[1]})`
+            // console.log(res)
+            // molecules[0].drawHover(res)
+
         }
     }
 
+    useEffect(() => {
+        iris_module().then(async (Module) => {
+            const map_response = await maps[0].getMap()
+            const moleculeData = await molecules[0].getAtoms()
+            let map_data = new Uint8Array(map_response.data.result.mapData)
+
+            const molecule_name = `${molecules[0].name}.pdb`
+            const map_name = `${maps[0].name}.map`
+            Module['FS_createDataFile']('/', map_name, map_data, true, true, true)
+            Module['FS_createDataFile']('/', molecule_name, moleculeData, true, true, true)
+
+            let backend_call = Module.calculate_single_pdb(molecule_name, map_name, false);
```

Comment:
What kind of object is `backend_call`? Did you define it using `value_object` or `class` in the bindings? I need to know because if it's the latter we need to call `delete()` since C++ object handles must be explicitly deleted in JS.

Reply:
The return type is a `MultiResidueBinding` which is created with a value_object.

Here is the code:
```
EMSCRIPTEN_BINDINGS(iris_module)
{

  value_object<ResidueResult>("ResidueResults")
      .field("name", &ResidueResult::name)
      .field("value", &ResidueResult::value)
      .field("seqnum", &ResidueResult::seqnum)
      .field("metric", &ResidueResult::metric)
      .field("type", &ResidueResult::type);

  register_vector<ResidueResult>("vector<ResidueResults>");

  value_object<ChainResult>("ChainResults")
      .field("chain", &ChainResult::chain)
      .field("results", &ChainResult::results);

  register_vector<ChainResult>("vector<ChainResults>");
  register_vector<std::string>("vector<std::string>");

  value_object<ResultsBinding>("ResultsBinding")
      .field("result", &ResultsBinding::result)
      .field("chain_labels", &ResultsBinding::chain_labels)
      .field("file_name", &ResultsBinding::file_name);

  register_vector<ResultsBinding>("vector<ResultsBinding>");
  value_object<MultiResultsBinding>("MultiResultsBinding")
      .field("results", &MultiResultsBinding::results);

  function("calculate_single_pdb", &calculate_single_pdb);
  function("calculate_multi_pdb", &calculate_multi_pdb);

}
```

Reply:
https://github.com/Iris-Validation/iris-validation-backend/blob/master/wasm/src/iris-bind.cpp

Reply:
Thanks for the info, merging this now...